### DISCUSSION TITLE
No known class method for selector `setupSnapshot` in Obj-C project

### DISCUSSION THREAD

**Discussion by mciarlo** [fastlane/fastlane#16826] [Q&A] [answered]
# Issue Checklist
- [x] I followed the [Getting Started guide for iOS](https://docs.fastlane.tools/getting-started/ios/setup/)
- [x] I tried the troubleshooting methods [here](https://github.com/fastlane/fastlane/issues/1657)
- [x] I re-created the Target and re-tried the configuration

# Description
After following the installation and troubleshooting guides, running `fastlane snapshot` results in `Exit status: 75` with the details: `no known class method for selector 'setupSnapshot:'`

# Fastlane Env

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | 10.14.6                                                 |
| Ruby                        | 2.7.1                                                   |
| Bundler?                    | false                                                   |
| Git                         | git version 2.21.1 (Apple Git-122.3)                    |
| Installation Source         | /usr/local/Cellar/fastlane/2.151.2/libexec/bin/fastlane |
| Host                        | Mac OS X 10.14.6 (18G5033)                              |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.7.1_2/lib                      |
| OpenSSL Version             | OpenSSL 1.1.1f  31 Mar 2020                             |
| Is contained                | false                                                   |
| Is homebrew                 | false                                                   |
| Is installed via Fabric.app | false                                                   |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/             |
| Xcode Version               | 11.3.1                                                  |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.151.2 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

*generated on:* **2020-07-10**
</details>

**Comment by mciarlo** [comment, marked as answer]
I resolved this issue. It appears the documentation is incorrect. The proper initialization method is `[Snapshot setupSnapshot:app waitForAnimations:NO];`. The documentation currently has `[Snapshot setupSnapshot:app];`

**Reply by joshdholtz** [reply]
Ah, thanks for reporting the solution! I will create an issue so this gets fixed 🙂

**Reply by joshdholtz** [reply]
Issue created: https://github.com/fastlane/docs/issues/942

**Reply by mciarlo** [reply]
You're welcome! I'd have tried submitting a pull request but I'm quite new to OSS. Glad I was helpful!

**Reply by rogerluan** [reply]
I created a PR to fix this :) 
https://github.com/fastlane/docs/pull/993
